models/articles-model.js

```diff
@@ -16,21 +16,36 @@ exports.selectArticles = (articleId) => {
     });
 };
 
-exports.selectAllArticles = () => {
-  return db
-    .query(
-      `
-    SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COALESCE(COUNT(comments.article_id), 0) AS comment_count
-    FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC
-  `
-    )
-    .then((result) => {
-      const articles = result.rows;
-      return articles;
-    });
+exports.selectAllArticles = (sortBy, order) => {
+  let baseQuery = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COALESCE(COUNT(comments.article_id), 0) AS comment_count
+  FROM articles
+  LEFT JOIN comments ON articles.article_id = comments.article_id
+  GROUP BY articles.article_id`;
+
+  const validSortBy = [
+    "author",
+    "title",
+    "article_id",
+    "topic",
+    "created_at",
+    "votes",
+    "comment_count",
```

Comment:
You already have this in your controller in an attempt to whitelist, if it got through the controller is it needed here? You only need to check your request parameters in one location. As previously mentioned, while it is unlikely your data will change, it _could_ so how could you make this array more programmatic?